lib/OpenQA/Shared/Controller/Auth.pm

```diff
@@ -181,7 +181,7 @@ sub _valid_hmac ($self, $hash, $request, $build_tx_timestamp, $timestamp, $api_k
     return 0 unless $api_key->secret;
 
     my $base_url = $self->app->config->{global}->{base_url};
-    my $base_path = $base_url ? Mojo::URL->new($base_url)->path : '';
+    my $base_path = $base_url ? Mojo::URL->new($base_url)->path->leading_slash(0) : '';
```

Comment:
According to the issue, it is the trailing slash for the `base_url` which became the leading slash in the `base_path`.

I wasn't able to find a good place where this code path is actually tested, although I was able to reproduce the issue by manually specifying different base url in the controller code.

Reply:
You could test this in `03-auth.t`. As I mentioned, for testing this detailed behavior it would likely best to extract this small piece of code into a function and just invoke that directly *somewhere* from the tests.

However, your explanation actually sounds good enough :-)